checkov/terraform/graph_builder/foreach_handler.py

```diff
@@ -107,5 +115,38 @@ def _handle_static_statement(self, block_index: int) -> Optional[list[str] | dic
             return self._handle_static_count_statement(count_statement)
         return
 
-    def _create_new_foreach_resources(self, block_index: int, foreach_statement: list[str] | dict[str, Any]) -> None:
+    def _handle_dynamic_statement(self, blocks_to_render: list[int]) -> dict[int, Optional[list[str] | dict[str, Any] | int]]:
+        rendered_statements_by_idx: dict[int, Optional[list[str] | dict[str, Any] | int]] = {}
+        sub_graph = self._build_sub_graph(blocks_to_render)
+        self._render_sub_graph(sub_graph, blocks_to_render)
+        for block_idx in blocks_to_render:
+            if not self._is_static_statement(block_idx, sub_graph):
+                rendered_statements_by_idx[block_idx] = None
+            else:
+                rendered_statements_by_idx[block_idx] = self._handle_static_statement(block_idx, sub_graph)
```

Comment:
kind of wierd that a function called `_handle_dynamic_statement` is calling `_handle_static_statement`.
This implies this function is more like a general `_handle_statement`, and that handling a dynamic statement is basically the function - `lambda x: retrun None`

Reply:
First, we are running over all the resources and getting the statics -> all the None is probably a dynamic -> building a sub-graph for all the dynamic and rendering them -> need to validate if now they are static - this is why we are checking static in the dynamic flow

Reply:
ok :)